openpype/pipeline/delivery.py

```diff
@@ -162,8 +164,48 @@ def deliver_single_file(
     if not os.path.exists(delivery_folder):
         os.makedirs(delivery_folder)
 
-    log.debug("Copying single: {} -> {}".format(src_path, delivery_path))
-    _copy_file(src_path, delivery_path)
+    if has_renumbered_frame:
+        src_dir = os.path.dirname(src_path)
+        src_filepaths = []
+        for repre_file in repre["files"]:
+            src_path = anatomy.fill_root(repre_file["path"])
+            base_path = os.path.basename(src_path)
+            src_filepaths.append(base_path)
+        if not src_filepaths:
+            msg = "Source files not found, cannot find collection"
+            report_items[msg].append(src_path)
+            log.warning("{} <{}>".format(msg, src_filepaths))
+            return report_items, 0
+
+        src_collections, remainders = clique.assemble(src_filepaths)
+        src_collection = src_collections[0]
+        src_indexes = list(src_collection.indexes)
+
+        dst_filepaths = []
+        for index in src_indexes:
+            template_data = copy.deepcopy(anatomy_data)
+            template_data["frame"] = index
+            template_obj = anatomy.templates_obj["delivery"][template_name]
+            template_filled = template_obj.format_strict(
+                template_data
+            )
+            dst_filepaths.append(template_filled)
+
+        dst_collection = clique.assemble(dst_filepaths)[0][0]
+        relative = not has_renumbered_frame
+        dst_collection = shift_collection(
+            dst_collection, renumber_frame, relative=relative)
+
+        for src_file_name, dst in zip(src_collection, dst_collection):
+            src_path = os.path.join(src_dir, src_file_name)
+            delivery_path = os.path.join(delivery_folder, dst)
+            log.debug("Copying single: {} -> {}".format(
+                src_path, delivery_path))
+            _copy_file(src_path, delivery_path)
+    else:
+        log.debug("Copying single: {} -> {}".format(src_path, delivery_path))
+        _copy_file(src_path, delivery_path)
+
```

Comment:
Single file cannot be reframed
```suggestion
    log.debug("Copying single: {} -> {}".format(src_path, delivery_path))
    _copy_file(src_path, delivery_path)
```

Reply:
but i have tried with `deliver_sequence_file` codes. it seems that the delivery action only works with `deliver_single_file`

Reply:
you can scroll up to see all the changes requested, it seems `deliver_sequence` is deprecated for some reasons. At the early commit I set it through `deliver_sequence`, the delivery action won't work. And it works after I set up the delivery action into `delivery_single_file` in the later commit.

Reply:
But `delivery_single_file` by design cannot handle renumbering. Single file has no information about frame.

Reply:
> But `delivery_single_file` by design cannot handle renumbering. Single file has no information about frame.

this is something i find really strange at first but this is something true. If you remove all the renumber args in the `deliver_single_files `and just add the args into `deliver_sequence`, and you run tray to do the delivery action you find out `deliver_sequence `doesn't renumber any frame.  you can try the code with this commit: https://github.com/ynput/OpenPype/pull/5024/commits/8030ee9c71fcbad57802aff9f8296569e8c00d4d and you will know.

Reply:
Yes, but that should not be handled in this function. You have to change frame fill logic in `openpype/plugins/load/delivery.py`.